Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,458 @@
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Single array approach to add all Setters/Getters, so we don't have to deal with SO much boilerplate repetition
+    // Could be useful for other extensions... Thinking of making an online tool to generate this file from json data some day
+    // 0 function name, 1 description, 2 icon name, 3 param property type, 4 hasModifier
+    [
+      [
+        'BBText',
+        'BBCode formatted text',
+        'textAlign24',
+        'string',
+        'BBCode text',
+      ],
+      ['Color', 'base color', 'color24', 'color', 'Color (Hex)'],
+      ['Opacity', 'base opacity', 'opacity24', 'number', 'Opacity (0-255)'],
+      ['FontSize', 'base font size', 'characterSize24', 'number', 'Font size'],
+      ['FontFamily', 'base font family', 'font24', 'string', 'Font family'],
+      [
+        'Alignment',
+        'text alignment',
+        'textAlign24',
+        'stringWithSelector',
+        'Alignment',
+        `["left", "right", "center"]`,
+      ],
+      [
+        'WrappingWidth',
+        'wrapping width',
+        'scaleWidth24',
+        'number',
+        'Wrapping width',
+      ],
+    ].forEach(p => {
+      // Add all the generic GETTERS
+      if (p[3] === 'number') {
+        object
+          .addExpression(
+            `Get${p[0]}`,
+            _(`Get ${p[1]}`),
+            _(`Get ${p[1]}`),
+            _(''),
+            '',
+            `res/actions/${p[2]}.png`,
+            `res/actions/${p[2]}.png`
+          )
+          .addParameter('object', _('BBText object'), 'BBText', false)
+          .getCodeExtraInformation()
+          .setFunctionName(`get${p[0]}`);
+      } else {
+        object
+          .addStrExpression(
+            `Get${p[0]}`,
+            _(`Get ${p[1]}`),
+            _(`Get ${p[1]}`),
+            _(''),
+            '',
+            `res/actions/${p[2]}.png`,
+            `res/actions/${p[2]}.png`
+          )
+          .addParameter('object', _('BBText object'), 'BBText', false)
+          .getCodeExtraInformation()
+          .setFunctionName(`get${p[0]}`);
+      }
+
+      // Add all the generic SETTERS
+      if (p[3] === 'number' || p[3] === 'string') {
+        object
+          .addAction(
+            `Set${p[0]}`,
+            _(`Set ${p[1]}`),
+            _(`Set ${p[1]}`),
+            _(`Set ${p[1]} of _PARAM0_ to _PARAM1_ _PARAM2_`),
+            '',
+            `res/actions/${p[2]}.png`,
+            `res/actions/${p[2]}.png`
+          )
+          .addParameter('object', _('BBText object'), 'BBText', false)
+          .addParameter('operator', _("Modification's sign"), '', false)
+          .addParameter(
+            p[3] === 'number' ? 'expression' : 'string',
```

Comment:
Let's write "expression" rather than "number" then and let's avoid this whole ternary condition (we can just write p[3]) instead.
"expression" is maybe not a great name - number would be better. But that's a change to be done rather in GDCore :)

Reply:
Well, instead of `p[3]`, I mean obvisouly something with an object like: `property.type`.